interactive_engine/src/executor/store/src/v2/api/partition_snapshot.rs

```diff
@@ -18,38 +18,63 @@ use crate::v2::api::types::{Vertex, Edge, EdgeRelation};
 use crate::v2::api::condition::Condition;
 use crate::v2::Result;
 
+/// Snapshot of a graph partition. All the interfaces should be thread-safe
 pub trait PartitionSnapshot: Send + Sync {
     type V: Vertex;
     type E: Edge;
 
+    /// Returns the vertex entity of given `vertex_id`, properties are filtered
+    /// by the `property_ids` optionally.
+    ///
+    /// If `label_id` is [`None`], all vertex labels will be searched and the
+    /// first match vertex will be returned.
     fn get_vertex(
         &self,
         vertex_id: VertexId,
         label_id: Option<LabelId>,
         property_ids: Option<&Vec<PropertyId>>,
     ) -> Result<Option<Self::V>>;
 
+    /// Returns the edge entity of given `edge_id`, properties are filtered
+    /// by the `property_ids` optionally.
+    ///
+    /// If `edge_relation` is [`None`], all edge relations will be searched and
+    /// the first match edge will be returned.
     fn get_edge(
         &self,
         edge_id: EdgeId,
         edge_relation: Option<&EdgeRelation>,
         property_ids: Option<&Vec<PropertyId>>,
     ) -> Result<Option<Self::E>>;
 
+
+    /// Returns all vertices, filtered by `label_id` and `condition`
+    /// optionally.
+    ///
+    /// Properties of the vertices are filtered by the `property_ids`
+    /// optionally.
     fn scan_vertex(
         &self,
         label_id: Option<LabelId>,
         condition: Option<&Condition>,
         property_ids: Option<&Vec<PropertyId>>,
     ) -> Result<Records<Self::V>>;
 
+    /// Returns all edges, filtered by `edge_relation` and `condition`
+    /// optionally.
+    ///
+    /// Properties of the edges are filtered by the `property_ids` optionally.
     fn scan_edge(
         &self,
         edge_relation: Option<&EdgeRelation>,
         condition: Option<&Condition>,
         property_ids: Option<&Vec<PropertyId>>,
     ) -> Result<Records<Self::E>>;
 
+    /// Returns out edges of vertex `vertex_id`, filtered by `label_id` and
+    /// `condition` optionally.
+    ///
+    /// Properties of the edges are filtered by the `property_ids` optionally.
     fn get_out_edges(
```

Comment:
Should get_out_edge have label_id and condition as its input params?

Reply:
line 81~82